### PR TITLE
The Great Saiyan Race

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

<details>
When I started this joke PR it was before the news broke that Akira Toriyama had passed away, it wasn't really intended as any kind of memorial or remembrance. Hard to know what to say about losing someone who was such a massive influence to an enormous amount of modern culture except that it's obviously devastating.
Anyway that said, despite the climate under which this is now unintentionally posted, don't take this PR too seriously it's just (lovingly) for fun.
</details>

image

This PR adds a new species to the game: the mighty Saiyan warriors of planet Vegeta.
A species of born fighters and martial artists, Saiyans love battle and are feared throughout the galaxy for their practice of hurling their children in single-occupant pods to distant planets where the babies will destroy all present civilisation and the parents can then sell the unoccupied planet on the galactic market at cut-throat rates.

Due to their natural understanding of martial arts, Saiyans are capable of projecting their "ki" (or "fighting spirit") as energy blasts from their hands, and of course have trained long enough to know how to fly (anyone could do this if they trained hard enough). Each also knows a signature technique based on the school of fighting they trained under, or taught to themselves.

Their fighting spirit asserts itself when facing strong combatants: a Saiyan who is reduced to near-death but survives without expiring will receive a power boost in order to ensure the fight is more even next time. They pay for this with their massive appetite, and aversion towards firearms (their muscled fingers are too large to operate precise human equipment).

One unique feature of Saiyans (and indeed, virtually the only visual tell separating them from Humans) are their ape-like tails, used for balance but also connected to their most fearsome inherent ability.
A tailed Saiyan exposed to moonlight will rapidly transform into an enormous ape-like creature capable of firing powerful energy blasts from its mouth. Removing the tail will revert it to its original form, and leave the humiliated Saiyan both disoriented and depowered.
For the purposes of SS13 we're just going to assume that anywhere in space is probably going to contain light from some kind of moon somewhere.

image

## Why It's Good For The Game

We've been doing all of this stuff like "multi-z" and "economy" and "" and "atmospheric simulation" and with each change it brings us further and further from our roots, from what the BYOND engine was really made for. Namely: anime fangames made by and for pre-teens. Specifically where they can play as Goku.
I don't technically qualify as being one of those any more but I hope that adding this new species to the game can keep us humble and remind us of where we came from.

## Changelog

:cl:
add: You can now play as one of the galaxy's mightiest warriors
/:cl:

### CODE REVIEW

File: code/modules/mob/living/basic/farm_animals/gorilla/gorilla.dm [hunk 1]
```diff
@@ -159,4 +159,67 @@
...
+	corpse.death()
+	corpse.setBruteLoss(corpse.maxHealth, TRUE, TRUE)
+	return ..()
+
+/// Cut off his tail! It's the only way!
+/mob/living/basic/gorilla/saiyan/proc/check_tail_sever(mob/living/target, obj/item/weapon, mob/attacker, proximity_flag, click_parameters)
+	SIGNAL_HANDLER
+	if (!proximity_flag || weapon.force < 5 || weapon.get_sharpness() != SHARP_EDGED)
+		return
+	to_chat(world, "trying it")
```
Comment (mc-oofert) [inline code/modules/mob/living/basic/farm_animals/gorilla/gorilla.dm:199]: funny

**Comment by IndieanaJones** [comment]
HEY ITS ME GOKU

**Comment by Iamgoofball** [comment]
finally we can pay respects to byond's cultural heritage

**Comment by Jacquerel** [comment]
These tests pass when run locally 😔

**Review by Wallemations — APPROVED** [review]
o7

**Comment by CRITAWAKETS** [comment]
I think Ozaru form should be fucking gigantic for the bit.

**Comment by Jacquerel** [comment]
> I think Ozaru form should be fucking gigantic for the bit.

I don't like scaling up sprites and couldnt be bothered to make a new 96x96 sprite

File: code/modules/mob/living/carbon/human/species_types/saiyan.dm [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	vegeta.saiyan_boost(multiplier = 5)
+	vegeta.clear_mood_event(SAIYAN_TAIL_MOOD)
+
+/// If your tail is restored you return to original strength
+/datum/species/saiyan/proc/on_tail_removed(mob/living/vegeta, obj/item/organ/tail)
+	SIGNAL_HANDLER
+	if (!istype(tail, /obj/item/organ/external/tail/monkey/saiyan))
+		return
+	to_chat(vegeta, span_boldwarning("No! Your tail!!"))
+	vegeta.saiyan_boost(multiplier = -5)
```
Comment (ATH1909) [inline code/modules/mob/living/carbon/human/species_types/saiyan.dm:113]: won't a negative multiplier cause jank interactions?

**Comment by EricZilla** [comment]
Wanted to fool around with it, spawned as captain used kaioken and deleted myself, A+

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -2662,3 +2662,37 @@ GLOBAL_LIST_EMPTY(fire_appearances)
...
+#define DAMAGE_MULT 0.1
+#define SPEED_BOOST 0.1
+#define HEALTH_BOOST 5
+
+/// Generally get stronger, as a saiyan would (or weaker if we pass a negative multiplier)
+/mob/living/proc/saiyan_boost(multiplier = 1)
+	maxHealth += HEALTH_BOOST * multiplier // Fuck knows if this actually does anything
+	var/datum/action/cooldown/mob_cooldown/ki_blast/blast = locate() in actions
+	if (!isnull(blast))
+		blast.damage_modifier += DAMAGE_MULT * multiplier
```
Comment (ATH1909) [inline code/modules/mob/living/living.dm:2687]: do you have to dismiss and resummon your "gun" to update this?

File: code/modules/mob/living/living.dm [hunk 4]
```diff
@@ -2662,3 +2662,37 @@ GLOBAL_LIST_EMPTY(fire_appearances)
...
+	boost_movespeed -= SPEED_BOOST * multiplier
+	add_or_update_variable_movespeed_modifier(/datum/movespeed_modifier/saiyan_speed, TRUE, boost_movespeed)
+
+	var/added_damage = DAMAGE_BOOST * multiplier
+	melee_damage_lower += added_damage
+	melee_damage_upper += added_damage
```
Comment (ATH1909) [inline code/modules/mob/living/living.dm:2693]: does this work for carbons?

Comment (Jacquerel) [reply code/modules/mob/living/living.dm:2693]: Carbons have an extension of this proc which updates their body parts

File: code/modules/mob/living/living.dm (hunk 3 above)
Comment (Jacquerel) [reply code/modules/mob/living/living.dm:2687]: Yes but as you would have to enter hard crit in between you would do this automatically via dropping them

**Comment by Verbalase** [comment]
“It’s that all you got, Frieza?”

**Pull request closed without merging**